tests/try_alloc_with.rs

```diff
@@ -0,0 +1,77 @@
+// All of these try_alloc_with tests will fail with "fatal runtime error: stack overflow" unless LLVM
+// manages to optimize the stack writes away.
+//
+// We only run them when debug_assertions are not set, as we expect them to fail outside release
+// mode.
+
+use bumpalo::Bump;
+
+#[test]
+#[cfg_attr(debug_assertions, ignore)]
+#[should_panic(expected = "out of memory")]
+fn try_alloc_with_oom() {
+    todo!()
+}
```

Comment:
I'm not too sure I can actually test this. Rust refuses to compile extremely large slices, at least on my PC:

```
error: the type `[u8; 184467440737095516]` is too big for the current architecture
```

Reply:
I found the implementation in `try_alloc.rs` 😅
I'll see how I can adapt that.